### PR TITLE
9550-android_infrastructure_code

### PR DESCRIPTION
**Pull request by dominicmarmionswrve**
9550-android_infrastructure_code
https://swrvedev.jira.com/browse/SWRVE-9550

test changes:
https://github.com/Swrve/swrve-sdk/pull/888

### CODE REVIEW

**Comment by dominicmarmionswrve** [comment]
@Sergio-Mira can you take a look please?

File: SwrveSDK/src/main/java/com/swrve/sdk/ISwrveBase.java [hunk 1]
```diff
@@ -110,4 +110,7 @@
     Context getContext();
 
     C getConfig();
+
+    void geofenceCrossed(String campaignId, String geofenceId, String trigger);
```
Comment (Sergio-Mira) [inline SwrveSDK/src/main/java/com/swrve/sdk/ISwrveBase.java]: ~ ~ ~ ~ onGeofenceCrossed (like onDestroy, etc?)

File: SwrveSDK/src/main/java/com/swrve/sdk/locationcampaigns/LocationCampaignEngageReceiver.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import com.swrve.sdk.SwrveSDKBase;
+
+public class LocationCampaignEngageReceiver extends BroadcastReceiver {
+
+    private static final String LOG_TAG = "SwrveLocation";
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        FilterableNotification filterableNotification = intent.getParcelableExtra("notification");
+        if (intent.getParcelableExtra("notification") == null || SwrveHelper.isNullOrEmpty(filterableNotification.getData())) {
```
Comment (Sergio-Mira) [inline SwrveSDK/src/main/java/com/swrve/sdk/locationcampaigns/LocationCampaignEngageReceiver.java]: You can use filterableNotification here

File: SwrveSDK/src/main/java/com/swrve/sdk/ISwrveBase.java (hunk 1 above)
Comment (dominicmarmionswrve) [inline SwrveSDK/src/main/java/com/swrve/sdk/ISwrveBase.java]: onGeofenceCrossed sounds good.

**Comment by Sergio-Mira** [comment]
:rocket:  (will be harder to not distribute this by default if that is the idea, we'll need x2 flavours I guess?)

File: SwrveSDK/src/main/java/com/swrve/sdk/locationcampaigns/LocationCampaignEngageReceiver.java (hunk 2 above)
Comment (dominicmarmionswrve) [inline SwrveSDK/src/main/java/com/swrve/sdk/locationcampaigns/LocationCampaignEngageReceiver.java]: :+1:

**Comment by dominicmarmionswrve** [comment]
Yeah, distributing these separately may be a challenge, but will tackle that after FA

**Pull request merged**